src/Compiler/AbstractIL/ilread.fs

```diff
@@ -2248,11 +2242,16 @@ and seekReadNestedTypeDefs (ctxt: ILMetadataReader) tidx =
 and seekReadInterfaceImpls (ctxt: ILMetadataReader) mdv numTypars tidx =
     seekReadIndexedRows (
         ctxt.getNumRows TableNames.InterfaceImpl,
-        seekReadInterfaceImplRow ctxt mdv,
-        (fun x -> x.TypeIdx),
-        simpleIndexCompare tidx,
+        id,
+        id,
+        (fun idx ->
+            let mutable addr = ctxt.rowAddr TableNames.InterfaceImpl idx
+            let _tidx = seekReadUntaggedIdx TableNames.TypeDef ctxt mdv &addr
+            simpleIndexCompare tidx _tidx),
         isSorted ctxt TableNames.InterfaceImpl,
-        (fun x -> (seekReadTypeDefOrRef ctxt numTypars AsObject [] x.IntfIdx), (ctxt.customAttrsReader_InterfaceImpl, x.IntImplIdx))
+        (fun idx ->
+            let intfIdx = seekReadInterfaceIdx ctxt mdv idx
+            seekReadTypeDefOrRef ctxt numTypars AsObject [] intfIdx, (ctxt.customAttrsReader_InterfaceImpl, idx))
```

Comment:
If this still reads the attributes applied on interface impls, then it's definitely good to get rid of  the anon from the signature!